### PR TITLE
feat(secrets): add vault secrets support

### PR DESCRIPTION
**Pull request by KathrynLewis**

### CODE REVIEW

File: pkg/secrets/vault.go [hunk 1]
```diff
@@ -2,23 +2,222 @@ package secrets
...
+	}
+
+	if vaultConfig.AuthMethod == "TOKEN" {
+		if token := os.Getenv("VAULT_TOKEN"); token == "" {
+			return fmt.Errorf("VAULT_TOKEN environment variable not set")
+		}
+	} else if vaultConfig.AuthMethod == "KUBERNETES" {
+		if vaultConfig.Path == "" || vaultConfig.Role == "" {
+			return fmt.Errorf("path and role both required for Kubernetes auth method")
+		}
```
Comment (ncknt) [inline pkg/secrets/vault.go]: shouldn't it be caught in the config validation?

File: pkg/secrets/vault.go [hunk 2]
```diff
@@ -2,23 +2,222 @@ package secrets
...
+	}
+	for _, element := range configs {
+		kv := strings.Split(element, ":")
+		if len(kv) < 2 {
+			return VaultSecret{}, fmt.Errorf("illegal format for key-value pair in %q: %s",
+				encryptedSecret, element)
+		}
+		switch kv[0] {
+		case "encrypted":
+			// do nothing
```
Comment (ncknt) [inline pkg/secrets/vault.go]: that's a bizarre parsing. Shouldn't it be already parsed before you determine the engine?

File: pkg/secrets/s3.go [hunk 3]
```diff
@@ -23,37 +23,37 @@ type S3Secret struct {
...
 }
 
 func (s3 *S3Decrypter) Decrypt() (string, error) {
-	s3Secret, err := parseS3SecretConfig(s3.secretConfig)
+	s3Secret, err := parseS3EncryptedSecret(s3.encryptedSecret)
 	if err != nil {
 		return "", err
 	}
-	secret, err := s3Secret.fetchSecret()
+	decrypted, err := s3Secret.fetchSecret()
```
Comment (ncknt) [inline pkg/secrets/s3.go]: can't you just `return s3Secret.fetchSecret()`?

File: pkg/secrets/s3.go [hunk 4]
```diff
@@ -67,7 +67,7 @@ func parseS3SecretConfig(secretConfig string) (S3Secret, error) {
 		case "k":
```
Comment (ncknt) [inline pkg/secrets/s3.go]: Same comment as for vault. Looks like we're parsing the token twice and repeat the code too. I'd expect `ParseToken()` to just return a ```{
engine string
params map[string]string
}```

File: pkg/secrets/s3.go (hunk 3 above)
Comment (KathrynLewis) [reply pkg/secrets/s3.go]: yes. yes i can.

File: pkg/secrets/vault.go (hunk 2 above)
Comment (jasonmcintosh) [reply pkg/secrets/vault.go]: Is this tied to the test case for "plainTextSecret"?  Where there are "secrets" that aren't encrypted?

**Pull request closed without merging**